src/librustdoc/html/render.rs

```diff
@@ -218,7 +219,7 @@ struct SharedContext {
 }
 
 impl SharedContext {
-    fn ensure_dir(&self, dst: &Path) -> Result<(), Error> {
+    pub fn ensure_dir(&self, dst: &Path) -> Result<(), Error> {
```

Comment:
Wouldn't this become an "unreachable pub", once we turn that warning on, due to the fact that `SharedContext` is `crate`?